pandas/plotting/_core.py

```diff
@@ -342,7 +342,13 @@ def _compute_plot_data(self):
                 label = 'None'
             data = data.to_frame(name=label)
 
-        numeric_data = data._convert(datetime=True)._get_numeric_data()
+        # GH16953, _convert is needed as fallback, for ``Series``
+        # with ``dtype == object``
+        data = data._convert(datetime=True, timedelta=True)
```

Comment:
did you take out the ``_convert`` and see, on what input does it fail?

Reply:
Yap i did, circleci fails at ``tests.plotting.test_series.test_valid_object_plot``, since the ``Series`` is of ``dtype == object`` (`s = Series(lrange(10), dtype=object)`, also see comment on your last requested changes).
Also i think doing a general `data._convert` is more sturdy, than just calling in if the `Series`is of type object (which would be the minimal change for the test to pass).

Reply:
ok I guess leave it in is ok. In general we *don't* automatically coerce object types (even if they are numbers) in other functions / places; we happen to do it in plotting for compat I think. Can you open a new issue for this to discuss (deprecating this).